jerry-core/jerry.c

```diff
@@ -1157,29 +1152,19 @@ jerry_invoke_function (bool is_invoke_as_constructor, /**< true - invoke functio
                                              args_count);
   }
 
-  if (ECMA_IS_VALUE_ERROR (call_completion))
-  {
-    /* unhandled exception during the function call */
-    is_successful = false;
-  }
-
-  if (retval_p != NULL)
-  {
-    *retval_p = call_completion;
-  }
-
-  return is_successful;
+  return call_completion;
 } /* jerry_invoke_function */
 
 /**
  * Construct new TypeError object
+ *
+ * @return TypeError object value
  */
-static void
-jerry_construct_type_error (jerry_value_t *retval_p) /**< [out] value with constructed
-                                                      *         TypeError object */
+static inline jerry_value_t __attr_always_inline___
+jerry_construct_type_error (void)
```

Comment:
inline again. I think inlining is important, but it is unclear to me how can we take advantage of that, so I would do it as a separate step. Such change would not even hurt the api.